### PR TITLE
feat: Primer theme foundation + shared useComposerText primitive

### PR DESCRIPTION
**Pull request by rmusser01**
## Summary

Foundation work for the WebUI chat-composer redesign. Phase 1 ships the full theme layer (Primer as default for new users, whitelabel-ready); Phase 2 (start) extracts the first shared hook primitive and refactors Playground to consume it.

Based on the design review at `tmp_dir/ui_kits/webui/composer-redesign.html` (V1/V3/V5 variants) and the decisions captured in the planning session.

### Phase 1 — Theme foundation

- **Primer theme** registered as a built-in alongside `default`, `solarized`, `nord`, `high-contrast`, `rose-pine`. Dark = phosphor cyan on near-black; light = "Parchment Primer" (warm cream + deep teal ink).
- **New-user migration**: one-shot `tldw:themeMigrationVersion` marker in `localStorage`. Fresh installs default to `primer`; users with any stored preset (including explicit `default`) keep theirs. Handles corrupted markers, quota-exceeded errors, Safari private mode.
- **JetBrains Mono self-hosted** at `apps/tldw-frontend/public/fonts/` (Regular / Medium / SemiBold .ttf) matching existing Inter / Space Grotesk pattern. No remote Google Fonts calls.
- **Theme-agnostic glow tokens** (`--glow-primary`, `--glow-accent`, `--glow-success`) in `tailwind-shared.css`, derived from the active theme's color tokens so variants reskin automatically under any registered theme.
- **YAML catalog entries** for `preferences.ui.composer_variant` (enum `v1`/`v3`/`v5`, user-editable) and `preferences.ui.theme_migration_version` (integer, platform-admin-only).
- **`Docs/User_Guides/Theming.md`** — whitelabel authoring guide with schema reference.

### Phase 2 (start) — Shared composer scaffold

- **`components/Chat/composer/types.ts`** — `ChatComposerSurface`, `ChatComposerVariant`, `ChatComposerDoc`, `ChatComposerSubmitPayload` (superset of Playground's and Sidepanel's current submit shapes, per phase-0 spike), `ChatComposerContext`.
- **`useComposerText` primitive** — bundles form state (`useSimpleForm`) + draft persistence (`useDraftPersistence`, keyed per surface) + mobile-aware focus helper. Auto-resize stays caller-owned so Playground can size against its derived collapse-display value.
- **Playground's `useComposerInput` refactored** to wrap `useComposerText`. Four duplicated concerns (form, draft, sizing helpers, focus) consolidated behind one call; collapse/perf/handler overlay preserved. Diff is additive + substitutive, not a rewrite.

### Code review findings addressed

Reviewer raised four items on Phase 1; all folded in before this PR:
- Primer dark `textSubtle/elevated` contrast bumped to meet 3:1 floor; dedicated Primer elevated-surface contrast test added.
- Glow tokens renamed from Primer-specific (`--glow-cyan`/`--glow-amber`/`--glow-mint`) to semantic (`--glow-primary`/`--glow-accent`/`--glow-success`).
- `theme_migration_version` YAML `editable_by` tightened to `[platform_admin]`.
- Migration hardened: NaN/corrupt version markers trigger re-migration; `setItem` wrapped in try/catch for Safari private mode + quota errors.

## Test plan

- [x] Unit: `cd apps/packages/ui && npx vitest run src/themes/__tests__ src/components/Chat/composer/__tests__ src/components/Option/Playground/__tests__/PlaygroundHooks.jsx-extension.guard.test.ts` — 34 tests pass across 8 files.
- [x] Full theme suite — contrast baselines hold for all built-in themes including Primer.
- [x] Playground hook guard — `<React.Profiler` marker in refactored `useComposerInput.tsx` preserved.
- [x] ComposerToolbar + AttachmentsSummary + ComposerTextarea.idle-collapse regression tests pass after the `useComposerInput` refactor (21 tests).
- [ ] Manual smoke: start `apps/tldw-frontend` dev server, log in as a fresh user, confirm Primer theme applies. Switch to other themes, confirm reskin. Type in the Playground composer, confirm draft persists + auto-resize works.

## Out of scope for this PR

- Composer variants V1/V3/V5 (next PR — variants consume `useComposerContext` once fully populated).
- Sidepanel adoption of `useComposerText` (follow-up; Sidepanel's `form.tsx` is 3,700 lines and carved concern-by-concern).
- Submit-hook extraction (next phase; Playground already has `usePlaygroundSubmit` as the template).
- Rose-pine `textSubtle/elevated` 2.82 contrast (pre-existing, surfaced by my earlier audit-widening attempt; reverted so this PR stays scoped).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/packages/ui/src/hooks/useTheme.ts [hunk 1]
```diff
@@ -9,6 +9,11 @@ import { getThemeById, getDefaultTheme, getAllPresets } from "@/themes/presets"
 import { applyThemeTokens, clearThemeTokens } from "@/themes/apply-theme"
 import { buildAntdThemeConfig } from "@/themes/antd-theme"
 import { saveCustomTheme as saveCustomThemeFn, deleteCustomTheme as deleteCustomThemeFn } from "@/themes/custom-themes"
+import { migrateThemeUserPreferences } from "@/themes/user-preference-migration"
+
+// Run once on module import so the migration lands before useSetting hydrates
+// its in-memory state from localStorage on first render.
+migrateThemeUserPreferences()
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/hooks/useTheme.ts]: Moved the migration trigger out of the module-import side effect. THEME_PRESET_SETTING now runs migrateThemeUserPreferences() through a beforeGet hook in the settings registry, so the migration still happens before legacy localStorage hydration without firing during module import.

File: apps/packages/ui/src/themes/user-preference-migration.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    const rawVersion = window.localStorage.getItem(THEME_MIGRATION_VERSION_KEY)
+    // Coerce missing/corrupt values to 0 so migration runs; NaN >= N is always false.
+    const storedVersion = Number.parseInt(rawVersion ?? "0", 10)
+    const normalizedVersion = Number.isFinite(storedVersion) ? storedVersion : 0
+    if (normalizedVersion >= CURRENT_USER_PREFERENCE_MIGRATION) return
+
+    const storedPreset = window.localStorage.getItem(THEME_PRESET_KEY)
+    if (storedPreset === null) {
+      window.localStorage.setItem(THEME_PRESET_KEY, "primer")
+    }
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/themes/user-preference-migration.ts:56]: Adjusted the migration to be more conservative: a missing preset now seeds primer only when there is no other persisted local UI state, while explicit or legacy presets continue to be preserved. I also added regression coverage for both fresh-install and existing-user paths.

File: apps/packages/ui/src/components/Chat/composer/hooks/useComposerText.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    if (el.selectionStart === el.selectionEnd) {
+      const ua = typeof navigator !== "undefined" ? navigator.userAgent : ""
+      const isMobile =
+        /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(ua)
+      if (!isMobile) {
+        el.focus()
+      } else {
+        el.blur()
+      }
+    }
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/components/Chat/composer/hooks/useComposerText.ts:122]: I left this behavior as-is. useComposerText was extracted from the Playground's existing focus helper, where the mobile no-op avoids unwanted keyboard pop-ups when we restore focus after queue/send/notice actions. The Sidepanel intentionally keeps its own plain focus helper for unconditional focus, so this shared helper is preserving established Playground semantics rather than acting as a generic cross-surface focus utility.

File: Docs/User_Guides/Theming.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+To flip the default for *new* users to your brand:
+
+1. Bump `CURRENT_USER_PREFERENCE_MIGRATION` in [`user-preference-migration.ts`](../../apps/packages/ui/src/themes/user-preference-migration.ts).
+2. Extend the migration function to write your preset ID for the new version.
+
+Migration preserves existing user selections — only installs with no stored preset get the new default.
+
+## Composer variants × theme
+
+The chat-composer variants (V1 Terminal Stack, V3 Split Brief, V5 Radial Command) are **theme-agnostic**: they reference only CSS custom properties (`--color-primary`, `--glow-cyan`, etc.). Any registered theme reskins all three variants automatically.
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Docs/User_Guides/Theming.md]: Updated the theming guide to use --glow-primary so the docs match the semantic glow tokens defined in tailwind-shared.css.

File: apps/packages/ui/src/components/Chat/composer/__tests__/useComposerText.test.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      useComposerText({
+        draftKey: "tldw:test:inputprops",
+        textareaRef: createRef(),
+      })
+    )
+
+    const props = result.current.form.getInputProps("message")
+
+    expect(typeof props.onChange).toBe("function")
+    expect(props.value).toBe("")
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/components/Chat/composer/__tests__/useComposerText.test.tsx:93]: This thread landed on useComposerText.test.tsx, but the localStorage.setItem monkeypatch lives in themes/__tests__/user-preference-migration.test.ts. I fixed the actual cleanup there with try/finally so the original setter is always restored.

File: apps/packages/ui/src/components/Chat/composer/hooks/useComposerAttachments.ts [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+  /** Fired if base64 conversion throws. */
+  onImageReadError?: (error: unknown) => void
+}
+
+export interface UseComposerAttachmentsResult {
+  /** Ref for the hidden <input type="file" accept="image/*"> trigger. */
+  inputRef: React.RefObject<HTMLInputElement>
+  /** Ref for the hidden <input type="file"> document trigger. */
+  fileInputRef: React.RefObject<HTMLInputElement>
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/components/Chat/composer/hooks/useComposerAttachments.ts]: Widened the public attachment ref types to nullable RefObject<HTMLInputElement | null> so the interface matches the useRef(null) return shape exposed by the hook.

File: apps/packages/ui/src/components/Chat/composer/__tests__/useComposerAttachments.test.tsx [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+const makePdfFile = () =>
+  new File(["fake"], "doc.pdf", { type: "application/pdf" })
+const makeBlockedFile = () =>
+  // application/zip is in otherUnsupportedTypes in the real module
+  new File(["fake"], "archive.zip", { type: "application/zip" })
+
+const makeEvent = (file: File) =>
+  ({
+    target: { files: [file], value: "" },
+  }) as unknown as React.ChangeEvent<HTMLInputElement>
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/components/Chat/composer/__tests__/useComposerAttachments.test.tsx:20]: Seeded the fake input value with a non-empty file path so the reset assertion is observable and only passes if onFileInputChange clears the field.

File: apps/packages/ui/src/components/Chat/composer/types.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+ */
+export interface ChatComposerContext {
+  /** Which surface we're rendering for. */
+  surface: ChatComposerSurface
+
+  /** Variant chosen by the user's preference. */
+  variant: ChatComposerVariant
+
+  // TODO(phase-2): text, model, sources, attachments, commands, submit
+  // slices populated as each concern is lifted into the shared hook.
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/components/Chat/composer/types.ts]: Removed the TODO-style wording here so this no longer introduces an untracked TODO in changed code.

File: apps/packages/ui/src/themes/user-preference-migration.ts [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+ */
+export function migrateThemeUserPreferences(): void {
+  if (typeof window === "undefined") return
+
+  try {
+    const rawVersion = window.localStorage.getItem(THEME_MIGRATION_VERSION_KEY)
+    // Coerce missing/corrupt values to 0 so migration runs; NaN >= N is always false.
+    const storedVersion = Number.parseInt(rawVersion ?? "0", 10)
+    const normalizedVersion = Number.isFinite(storedVersion) ? storedVersion : 0
+    if (normalizedVersion >= CURRENT_USER_PREFERENCE_MIGRATION) return
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/themes/user-preference-migration.ts:45]: Tightened version parsing so only fully numeric migration markers are accepted. Partially numeric values like 1foo now fall back to 0, rerun the migration, and are covered by a regression test.

File: apps/packages/ui/src/components/Sidepanel/Chat/form.tsx [hunk 10]
```diff
@@ -310,11 +298,15 @@ export const SidepanelForm = ({
...
+  // Shared primitive: form state + draft persistence.
+  // See apps/packages/ui/src/components/Chat/composer/hooks/useComposerText.ts.
+  // Sidepanel intentionally keeps its own local `textAreaFocus` below (plain
+  // .focus() without the mobile blur heuristic) to preserve exact behavior —
+  // adopt the primitive's `textAreaFocus` in a follow-up if desired.
+  const { form, draftSaved } = useComposerText({
+    draftKey: storageKey,
+    textareaRef,
+    isProMode
   })
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/components/Sidepanel/Chat/form.tsx]: Sidepanel now pulls clearDraft() from useComposerText and calls it after successful send and queue paths so persisted drafts are cleared with the rest of the composer state. I added a source contract assertion for that integration as well.

File: apps/packages/ui/src/themes/__tests__/user-preference-migration.test.ts [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+  it("does not throw if localStorage.setItem rejects (quota / Safari private)", () => {
+    const originalSetItem = window.localStorage.setItem
+    window.localStorage.setItem = () => {
+      throw new Error("QuotaExceededError")
+    }
+
+    expect(() => migrateThemeUserPreferences()).not.toThrow()
+
+    window.localStorage.setItem = originalSetItem
+  })
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/themes/__tests__/user-preference-migration.test.ts:129]: Wrapped the localStorage.setItem monkeypatch in try/finally so cleanup always runs even if the assertion fails.

File: apps/packages/ui/src/components/Chat/composer/types.ts [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+  // Sidepanel-only
+  uploadedFiles?: unknown[]
+  requestOverrides?: {
+    chatMode?: string
+    selectedModel?: string
+    selectedSystemPrompt?: string
+    toolChoice?: "auto" | "required" | "none"
+    useOCR?: boolean
+    webSearch?: boolean
+  }
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/components/Chat/composer/types.ts:56]: Updated requestOverrides to accept the nullable replay state we already persist in queued request snapshots, including selectedModel, selectedSystemPrompt, and toolChoice. I also added a type test covering the null case.
